### PR TITLE
[tests] Test case for multiple bindings with different sizes

### PR DESCRIPTION
**Pull request by andyleiserson**
**Connections**
Test case for https://github.com/gfx-rs/wgpu/issues/7359 (currently failing)

**Description**
This test creates two bind groups against the same bind group layout, in the same compute pass, but against two different shaders that have different binding sizes. The first has binding size 8, the second has binding size 4.

**Squash or Rebase?** Squash

**Checklist**

- [x] Run `cargo fmt`.
- [ ] Run `taplo format`.
- [x] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown`
- [x] Run `cargo xtask test` to run tests.
- [ ] If this contains user-facing changes, add a `CHANGELOG.md` entry.

### CODE REVIEW

File: tests/gpu-tests/bind_groups.rs [hunk 1]
```diff
@@ -1,4 +1,120 @@
...
+use std::num::NonZeroU64;
+
+use wgpu::{BufferUsages, PollType};
+use wgpu_test::{gpu_test, FailureCase, GpuTestConfiguration, TestParameters, TestingContext};
+
+/// Create two bind groups against the same bind group layout, in the same
+/// compute pass, but against two different shaders that have different binding
+/// sizes. The first has binding size 8, the second has binding size 4.
+///
+/// Regression test for https://github.com/gfx-rs/wgpu/issues/7359.
```
Comment (cwfitzgerald) [inline tests/gpu-tests/bind_groups.rs:10]: If this is a regression test, it probably should be in `gpu-tests/regression`

**Review by cwfitzgerald — APPROVED** [review]
Approved with nit

File: tests/gpu-tests/bind_groups.rs (hunk 1 above)
Comment (andyleiserson) [reply tests/gpu-tests/bind_groups.rs:10]: I did see that directory, I was on the fence whether this is better classified as a regression test or as a functional test for a corner case. My preference is to classify tests functionally, and call something a regression test only if it is a unique combination of functionality that defies other classification. But I'm happy to put it wherever you think it fits best in this project.

Comment (cwfitzgerald) [reply tests/gpu-tests/bind_groups.rs:10]: I also a bit torn about the regression test classification, so I think leaving this is fine, if you feel one way or another about it.
